Solution

Hladkyi03 opened the pull request:
[DEMO LINK](https://Hladkyi03.github.io/react_todo-app-with-api/)

OleziO commented on src/App.tsx:
```diff
@@ -1,24 +1,216 @@
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
```
Try to fix linter error
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/56752562/cbb61363-3bd3-43b0-a51d-47b6257555d0)

OleziO commented on src/App.tsx:
```diff
@@ -1,24 +1,216 @@
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length > 0 && (
```
```suggestion
          {todos.length && (
```

OleziO commented on src/App.tsx:
```diff
@@ -1,24 +1,216 @@
+          )}
+        </section>
+
+        {todos.length > 0 && (
```
```suggestion
        {todos.length && (
```

OleziO commented on src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,96 @@
+
+  const handleSetFilter = (filter: string) => {
+    switch (filter) {
+      case 'active':
```
Move all cases to the enum

OleziO commented on src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,96 @@
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {activeTodos}
+        {' '}
+        items left
```
```suggestion
        {`${activeTodos} items left`}
```

OleziO commented on src/components/Todo/TodoItem.tsx:
```diff
@@ -0,0 +1,210 @@
+
+  useEffect(() => {
+    setIsChecked(completed);
+  }, [todo]);
```
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/56752562/43b3dbf0-0e38-472c-ba08-6f5381cc190c)
Try to fix linter error

OleziO commented on src/components/Todo/TodoItem.tsx:
```diff
@@ -0,0 +1,210 @@
+/* eslint-disable no-console */
```
Remove this comment

OleziO commented on src/utils/fetchClient.ts:
```diff
@@ -0,0 +1,46 @@
+  });
+}
+
+// To have autocompletion and avoid mistypes
```
Remove all comments from the code.

Hladkyi03 commented:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/87299182/4a840631-4b72-4fa8-a23c-53e866d02fe5)
this happens when i change condition to `todos.length` instead of  `todos.length > 0`, so i didn't change it. Also i tried to fix both linter errors, i fixed 1 in TodoItem. The other one in App is unchangeble, bcus linter says to remove dependency array, i removed it and the whole program was broken, so i just left that array withoth values

anastasiiavorobiova commented on src/components/ErrorNotification/ErrorNotification.tsx:
```diff
@@ -0,0 +1,37 @@
+        data-cy="HideErrorButton"
+        type="button"
+        className="delete"
+        onClick={() => setErrorMessage('')}
```
```suggestion
        onClick={resetMessage}
```
Consider moving this logic from jsx

anastasiiavorobiova reviewed (approved):
Great work!
